Tasks/GooglePlayHalt/googleutil.ts

```diff
@@ -0,0 +1,268 @@
+}
+
+/**
+ * Gets information for the specified app and track
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function getTrack(edits: any, packageName: string, track: string): Promise<Track> {
+    tl.debug('Getting Track information');
+    const requestParameters: PackageParams = {
+        packageName: packageName,
+        track: track
+    };
+
+    tl.debug('Additional Parameters: ' + JSON.stringify(requestParameters));
+    const getTrack = await edits.tracks.get(requestParameters);
+    return getTrack.data;
+}
+
+/**
+ * Update a given release track with the given information
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @param {integer or [integers]} versionCode - version code returned from an apk call. will take either a number or a [number]
+ * @param {double} userFraction - for rollouting out a release to a track, it's the fraction of users to get update 1.0 is all users. 
+ *                                0 means the release's APKs will no longer be served to users, users who already have these APKs are unaffected.
+ * @param {releaseNotes} releaseNotes - optional release notes to be attached as part of the update
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function updateTrack(edits: any, packageName: string, track: string, versionCode: any, userFraction: number, releaseNotes?: ReleaseNotes[]): Promise<Track> {
+    tl.debug('Updating track');
+    const release: AndroidRelease = {
+        versionCodes: (typeof versionCode === 'number' ? [versionCode] : versionCode)
+    };
+
+    if (releaseNotes && releaseNotes.length > 0) {
+        tl.debug('Attaching release notes to the update');
+        release.releaseNotes = releaseNotes;
+    }
+
+    if (userFraction < 1.0 && userFraction > 0) {
+        release.userFraction = userFraction;
+        release.status = 'inProgress';
+    } else if (userFraction <= 0) {
+        // If userFraction is less than 0, consider it as a halting signal
+        // set this to 0.1 to avoid following errors:
+        // - Error: HALTED release must have fraction
+        // - Error: Rollout fraction must be greater than 0 and less than 1
+        // As the rollout is halted, so no actual impact of fraction will be introduced.
+        tl.debug('User fraction is less than 0% marking rollout "halted"');
+        release.userFraction = 0.1;
```

Comment:
Do we always need to set up userFraction to close-to-zero value? Could it make sense to set up 'status' for some cases only?
It would allow also to enhance it even further - to be able to resume roll out based out previous roll out fraction value.

Reply:
Add a boolean option to allow users to choose whether to keep userFraction unchanged.